src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -475,6 +488,129 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
+        [InlineData(TestScenario.Read)]
+        [InlineData(TestScenario.Copy)]
+        [InlineData(TestScenario.CopyAsync)]
+        [InlineData(TestScenario.ReadByte)]
+        [InlineData(TestScenario.ReadByteAsync)]
+        public async Task StreamTruncation_IsDetected(TestScenario scenario)
+        {
+            var buffer = new byte[16];
+            var source = Enumerable.Range(0, 64).Select(i => (byte)i).ToArray();
+            byte[] compressedData;
+            using (var compressed = new MemoryStream())
+            using (var gzip = new GZipStream(compressed, CompressionMode.Compress))
+            {
+                foreach (var b in source)
+                {
+                    gzip.WriteByte(b);
+                }
+
+                gzip.Dispose();
+                compressedData = compressed.ToArray();
+            }
+
+            for (var i = 1; i <= compressedData.Length; i += 1)
+            {
+                var expectException = i < compressedData.Length;
+                using (var compressedStream = new MemoryStream(compressedData.Take(i).ToArray()))
+                {
+                    using (var s = new GZipStream(compressedStream, CompressionMode.Decompress))
+                    {
+                        var decompressedStream = new MemoryStream();
+
+                        try
+                        {
+                            switch (scenario)
+                            {
+                                case TestScenario.Copy:
+                                    s.CopyTo(decompressedStream);
+                                    break;
+
+                                case TestScenario.CopyAsync:
+                                    await s.CopyToAsync(decompressedStream);
+                                    break;
+
+                                case TestScenario.Read:
+                                    while (ZipFileTestBase.ReadAllBytes(s, buffer, 0, buffer.Length) != 0) { };
+                                    break;
+
+                                case TestScenario.ReadAsync:
+                                    while (await ZipFileTestBase.ReadAllBytesAsync(s, buffer, 0, buffer.Length) != 0) { };
+                                    break;
+
+                                case TestScenario.ReadByte:
+                                    while (s.ReadByte() != -1) { }
+                                    break;
+
+                                case TestScenario.ReadByteAsync:
+                                    while (await s.ReadByteAsync() != -1) { }
+                                    break;
+                            }
+                        }
+                        catch (InvalidDataException e)
+                        {
+                            if (expectException)
+                                continue;
+
+                            throw new XunitException($"An unexpected error occured while decompressing data:{e}");
```

Comment:
```suggestion
                            throw new XunitException("An unexpected error occured while decompressing data.", e);
```
Use the overload that takes `innerException`.

Reply:
This constructor seems to be `protected` thus inaccessible.
```
protected XunitException(string userMessage, Exception innerException)
```
Error:
```
'XunitException.XunitException(string, Exception)' is inaccessible due to its protection level
```

Other tests use this pattern though: https://github.com/dotnet/runtime/blob/b4c746b7712e887af066d66d1ec777be6283f6f9/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexParserTests.netfx.cs#L32.

Please let me know if there is a better way.